cardano-cli/test/Test/Utilities.hs

```diff
@@ -1,27 +1,63 @@
-module Test.Utilities (diffVsGoldenFile) where
+module Test.Utilities
+  ( diffVsGoldenFile,
+    diffFileVsGoldenFile,
+  ) where
 
-import           Cardano.Prelude (ConvertText (..))
+import           Cardano.Prelude (ConvertText (..), HasCallStack)
 
 import           Control.Monad.IO.Class (MonadIO (liftIO))
 import           Data.Algorithm.Diff (PolyDiff (Both), getGroupedDiff)
 import           Data.Algorithm.DiffOutput (ppDiff)
 import           GHC.Stack (callStack)
+import qualified GHC.Stack as GHC
 import           Hedgehog (MonadTest)
+import qualified Hedgehog.Extras.Test as H
 import           Hedgehog.Extras.Test.Base (failMessage)
+import qualified Hedgehog.Internal.Property as H
+import qualified System.Directory as IO
+import qualified System.Environment as IO
+import qualified System.IO.Unsafe as IO
+
+-- | Whether the test should create the golden files if the file does ont exist.
```

Comment:
I don't think we should be creating golden files on the fly.  Why? Because a golden file is supposed to serve as a reference against unexpected _future_ changes. If we generate the files on the fly, the file is no longer really a "reference file" because whatever changes we may have made will be present in the "reference" file and our "actual" file. This will then give us a false positive in our test suite if we introduced unintended changes.